File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> hashmap;
        for (auto& n : nums)
            hashmap[n]++;

        int maxval = 1, maj = nums[0];
        for (auto& num : hashmap)
        {
            if (num.second > maxval)
            {
                maj = num.first;
                maxval = num.second;
            }
        }
        return maj;
    }
};

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        candidate = nums[0]
        count = 0
        for n in nums:
            if n == candidate:
                count += 1
            else:
                count -= 1
            
            if count == 0:
                candidate = n
                count += 1
        
        return candidate